In [1]:
import nltk
nltk.download('vader_lexicon')
from nltk.sentiment import SentimentIntensityAnalyzer

import numpy as np

[nltk_data] Downloading package vader_lexicon to
[nltk_data]     C:\Users\Abdullah\AppData\Roaming\nltk_data...
[nltk_data]   Package vader_lexicon is already up-to-date!


In [2]:
analyzer = SentimentIntensityAnalyzer()


## We start the analysis for each student

### Student 1

In [3]:
st1txt = "It was very long and comprehensive"

In [4]:
scores1 = analyzer.polarity_scores(st1txt)

In [5]:
print(scores1)

{'neg': 0.0, 'neu': 0.688, 'pos': 0.312, 'compound': 0.3102}


## Student 2

In [6]:
st2txt = "Human AI right <br /> I wish this was explained better since it was a bit confusing to understand initially. It took me some time to figure out that it means rights AI should keep in mind for systems that can directly affect humans and their rights."

In [7]:
scores2 = analyzer.polarity_scores(st2txt)

In [8]:
print(scores2)

{'neg': 0.038, 'neu': 0.848, 'pos': 0.113, 'compound': 0.5719}


## Student 3

In [9]:
st3txt = "This chapter discusses human rights in regard to AI implementation and is helpful for understanding those rights. In addition to the content of the chapter, the queries provide me with a good understanding of how AI can be used to protect human rights."

In [10]:
scores3 = analyzer.polarity_scores(st3txt)

In [11]:
print(scores3)

{'neg': 0.0, 'neu': 0.825, 'pos': 0.175, 'compound': 0.8074}


## Student 4

In [12]:
st4txt = "The chapter explains a small part of the impact of AI on basic human rights. It gives a give introduction information that can be easily extended later on."

In [13]:
scores4 = analyzer.polarity_scores(st4txt)

In [14]:
print(scores4)

{'neg': 0.0, 'neu': 0.912, 'pos': 0.088, 'compound': 0.34}


## Student 5

In [15]:
st5txt = "The questions were all partially understandable. Some questions were formulated too vaguely for me. I didn't quite understand what was supposed to be written. <br /> <br /> It was also extremely difficult to find good answers to the questions."

In [16]:
scores5 = analyzer.polarity_scores(st5txt)

In [17]:
print(scores5)

{'neg': 0.065, 'neu': 0.867, 'pos': 0.068, 'compound': 0.0276}


## Student 6

In [18]:
st6txt = "It is hard to comment on the usefulness of the chapter if the words are too vague."

In [19]:
scores6 = analyzer.polarity_scores(st6txt)

In [20]:
print(scores6)

{'neg': 0.147, 'neu': 0.737, 'pos': 0.116, 'compound': 0.1027}


## Student 7

In [21]:
st7txt = "This chapter is a bit difficult for me but the questions are very interesting."

In [22]:
scores7 = analyzer.polarity_scores(st7txt)

In [23]:
print(scores7)

{'neg': 0.105, 'neu': 0.657, 'pos': 0.238, 'compound': 0.5006}


## Student 8

In [24]:
st8txt = "This chapter was very helpful in understanding the aspects of AI on different human rights."

In [25]:
scores8 = analyzer.polarity_scores(st8txt)

In [26]:
print(scores8)

{'neg': 0.0, 'neu': 0.819, 'pos': 0.181, 'compound': 0.4754}


## Student 9

In [27]:
st9txt = "It was a useful, clear and easy to follow chapter. It is important to consider the implications of AI and the potential benefits and drawbacks it may have on society. Creating laws and conventions for AI can help shape a better system where AI can benefit everyone equally."

In [28]:
scores9 = analyzer.polarity_scores(st9txt)

In [29]:
print(scores9)

{'neg': 0.0, 'neu': 0.611, 'pos': 0.389, 'compound': 0.9666}


## Student 10

In [30]:
st10txt = "I found it quite hard to always find a simple link between the rights and AI"

In [31]:
scores10 = analyzer.polarity_scores(st10txt)

In [32]:
print(scores10)

{'neg': 0.115, 'neu': 0.885, 'pos': 0.0, 'compound': -0.1761}


## Student 11

In [33]:
st11txt = "The chapter was useful in understanding various policies regading AI."

In [34]:
scores11 = analyzer.polarity_scores(st11txt)

In [35]:
print(scores11)

{'neg': 0.0, 'neu': 0.756, 'pos': 0.244, 'compound': 0.4404}


## Student 12

In [39]:
# Student 12 skipped, maybe remove from hte avg
st12txt = ""

In [40]:
scores12 = analyzer.polarity_scores(st12txt)

In [41]:
print(scores12)

{'neg': 0.0, 'neu': 0.0, 'pos': 0.0, 'compound': 0.0}


## Student 13

In [42]:
st13txt = "Chapter 5 is all about a brief discussion about AI ethics in Human Rights. If someone wants to implement or develop an Artificial Intelligence System he/she must have to maintain the rules which must be followed by the ethical rules of AI. This chapter is divided into several articles which represent Human Rights rules to manage an AI system that is clear and concise."

In [43]:
scores13 = analyzer.polarity_scores(st13txt)

In [44]:
print(scores13)

{'neg': 0.0, 'neu': 0.868, 'pos': 0.132, 'compound': 0.8402}


## Student 14

In [46]:
st14txt = "Some kind of feedback for the student could be nice.<br /> somehow I found the first question hard to answer"

In [47]:
scores14 = analyzer.polarity_scores(st14txt)

In [48]:
print(scores14)

{'neg': 0.072, 'neu': 0.928, 'pos': 0.0, 'compound': -0.1027}


## Student 15

In [49]:
st15txt = "This chapter was very difficult and to some level unclear as relating human rights to ai can be somewhat abstract."

In [50]:
scores15 = analyzer.polarity_scores(st15txt)

In [51]:
print(scores15)

{'neg': 0.21, 'neu': 0.79, 'pos': 0.0, 'compound': -0.5849}


## Student 16

In [55]:
st16txt = "this was a very long chapter"

In [56]:
scores16 = analyzer.polarity_scores(st16txt)

In [57]:
print(scores16)

{'neg': 0.0, 'neu': 1.0, 'pos': 0.0, 'compound': 0.0}


In [58]:
# create a list of dictionaries to add
scores_list = [scores1, scores2, scores3, scores4, scores5, scores6, scores7, scores8,
               scores9, scores10, scores11, scores12, scores13, scores14, scores15, scores16]

# specify the key i need, in this case, compound
key = 'compound'

# create a list to store the values of each key that i will extract from each dict
values_list = []

# loop over the list of dictionaries and extract the values for the specific key
for d in scores_list:
    if key in d:
        values_list.append(d[key])

# calculate the average of the values:
average = np.mean(values_list)

print(average)

0.28245624999999996


# Week 5/ chapter 5 scores

### The average score for week one sentiment analysis is 0.282 (1 being the most positive, 0 neutral, and -1 being negative)
### But pleaes note that some questions had a compound score of 0 since they were just neutral.